include/pagmo/detail/custom_comparisons.hpp

```diff
@@ -90,7 +90,7 @@ inline bool equal_to_f(T a, T b)
 }
 
 // equal_to_vf than compares vectors of floating point types considering nan==nan
-template <typename T>
+template <typename T, detail::enable_if_is_floating_point<T> = 0>
```

Comment:
It's not gonna do anything different from the static assertion, and it is more opaque. If this was a template specialisation, you could use it to pick another specialisation after you have disabled the invalid one, but in this case there's only one implementation.